tests/test_core_to_contrib.py

```diff
@@ -757,6 +757,11 @@
         "airflow.providers.amazon.aws.operators.batch.AwsBatchOperator",
         "airflow.contrib.operators.awsbatch_operator.AWSBatchOperator",
     ),
+    # # Leave this out to avoid TypeError: Instance and class checks can only be used with @runtime protocols
+    # (
+    #     "airflow.providers.amazon.aws.hooks.batch_client.AwsBatchProtocol",
+    #     "airflow.contrib.operators.awsbatch_operator.BatchProtocol",
+    # ),
```

Comment:
Should we uncomment it? It would be nice to have this test for deprecation.

Reply:
Yes, if there is a fix for testing `Protocol` but the typing system is failing on the `issubclass` test, as noted in the comment.  So the comment is a placeholder to warn about avoiding this error, because I don't know an easy fix for it (yet).  I did some cursory research on this error and did not find an easy solution - I did try to apply an `@runtime` decorator but that raised other errors.  I didn't want to change up the existing test method for the sub-class check, but the changes implemented do use the subclass approach required.  It's just that a `Protocol` raises a `TypeError` in the `issubclass` method; i.e.
```python
>>> from typing_extensions import Protocol
>>> class ProtoSubclass(Protocol):
...     pass
... 
>>> issubclass(ProtoSubclass, Protocol)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/envs/airflow/lib/python3.6/site-packages/typing_extensions.py", line 1265, in __subclasscheck__
    raise TypeError("Instance and class checks can only be used with"
TypeError: Instance and class checks can only be used with @runtime protocols
>>> from typing_extensions import runtime
>>> @runtime
... class ProtoSubclass(Protocol):
...     pass
... 
>>> issubclass(ProtoSubclass, Protocol)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/envs/airflow/lib/python3.6/site-packages/typing_extensions.py", line 1265, in __subclasscheck__
    raise TypeError("Instance and class checks can only be used with"
TypeError: Instance and class checks can only be used with @runtime protocols
```

Reply:
There are some other issues with the `Protocol` implementation, e.g.
- https://github.com/python/typing/issues/562
- `Protocol` might be best avoided, but that's not the point of this PR

Reply:
Maybe a solution to the problem will arrive on SO at
- https://stackoverflow.com/questions/59461689/how-can-you-test-that-a-python-typing-protocol-is-a-subclass-of-another-protocol

Reply:
A solution to this might be to use `@runtime_checkable` decorators, so if that works, I'll push up the fix for this PR to uncomment that code noted above.

Reply:
That solution works, but another check requires the moved class to be instantiated and that incurs `TypeError: Protocols cannot be instantiated` so the test class needs a bunch of special methods for Protocols.

Reply:
This is fixed, although it was more complicated than expected.  The `Protocol` classes need special treatment in the class definitions and tests.